web/modules/weather_blocks/src/Plugin/Block/Test/WFOPromoBlock.php.test

```diff
@@ -32,16 +33,79 @@ final class WFOPromoBlockTest extends Base
     {
         $this->onLocationRoute();
 
+        $promo = (object) [
+            "field_wfo" => (object) [
+                "entity" => (object) [
+                    "name" => (object) [
+                        "value" => "WFO name",
+                    ],
+                    "field_wfo_code" => (object) [
+                        "value" => "WFO code",
+                    ],
+                ],
+            ],
+            "body" => "WFO information body",
+            "field_phone_number_opt" => "phone number entity",
+            "field_facebook_url" => "facebook url entity",
+            "field_twitter_url" => "twitter url entity",
+            "field_youtube_url" => "youtube url entity",
+        ];
+
+        $this->entityService
+            ->method("getLatestNodeFromWFO")
+            ->willReturn($promo);
+
+        $expected = [
+            "name" => "WFO name",
+            "code" => "WFO code",
+            "about" => "WFO information body",
+            "phone" => "phone number entity",
+            "social" => [
+                "facebook" => "facebook url entity",
+                "twitter" => "twitter url entity",
+                "youtube" => "youtube url entity",
+            ],
+        ];
+
+        $actual = $this->block->build();
+
+        $this->assertEquals($expected, $actual);
+    }
+
+    /**
+     * Test that the block returns WFO name and code if there is no WFO info
+     * @group unit
+     * @group block
+     * @group wfo-promo-block
+     */
+    public function testReturnsWFOIfNoPromo(): void
+    {
+        $this->onLocationRoute();
+
+        $promo = null;
+
+        $this->entityService->method("getLatestNodeFromWFO")->willReturn(null);
+
         // Setup fetching the actual node.
-        $story = $this->createStub(ContentEntityInterface::class);
+        $wfoTaxonomyTerm = $this->createStub(ContentEntityInterface::class);
 
-        $story->method("id")->willReturn(94);
+        $wfoName = $this->createStub(TypedDataInterface::class);
```

Comment:
Another moment where I'm like "how did he figure this out?"

Reply:
Lord, I don't remember. I just copied it from elsewhere in the tests.